src/main/java/liquibase/ext/opensearch/lockservice/OpenSearchLockService.java

```diff
@@ -0,0 +1,155 @@
+package liquibase.ext.opensearch.lockservice;
+
+import liquibase.Scope;
+import liquibase.database.Database;
+import liquibase.exception.DatabaseException;
+import liquibase.ext.opensearch.database.OpenSearchConnection;
+import liquibase.ext.opensearch.database.OpenSearchLiquibaseDatabase;
+import liquibase.lockservice.DatabaseChangeLogLock;
+import liquibase.logging.Logger;
+import liquibase.nosql.lockservice.AbstractNoSqlLockService;
+import liquibase.util.NetUtil;
+import org.opensearch.client.opensearch.OpenSearchClient;
+import org.opensearch.client.opensearch._types.mapping.DateProperty;
+import org.opensearch.client.opensearch._types.mapping.KeywordProperty;
+import org.opensearch.client.opensearch._types.mapping.Property;
+import org.opensearch.client.opensearch._types.mapping.TextProperty;
+import org.opensearch.client.opensearch.core.search.Hit;
+import org.opensearch.client.opensearch.indices.PutMappingRequest;
+import org.opensearch.client.transport.httpclient5.ResponseException;
+
+import java.io.IOException;
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class OpenSearchLockService extends AbstractNoSqlLockService<OpenSearchLiquibaseDatabase> {
+
+    /**
+     * Magic ID: there will only ever be 0 or 1 entries in the lock index (we use the `create` API to ensure that it fails if the ID already exists)
```

Comment:
@reta: i don't see the advantage of that? now i'm just creating/deleting/querying a document to see if i have a lock - otherwise i have to read/write the metadata (of an index which would then never contain any documents). for debugging this might also be harder since i presume that more people will be able to quickly query the documents (just run `GET /databasechangeloglock/_search`) than the metadata (`GET /databasechangeloglock/_mapping` and then look for `_meta`)

Reply:
> @reta: i don't see the advantage of that?

@rursprung so what would happen if there are multiple indices are being migrated? There is one single index (`getDatabaseChangeLogLockTableName()`) that creates documents based on `final var lockEntry = new DatabaseChangeLogLock(LOCK_ENTRY_ID, new Date(), getLockedBy());`, with multiple migrations going on we could get conflicts (since `Date` still has some precision) over non-related indices, right?